SteppenWolfFenrir/natours
# Responsive design

## Summary
This PR implements responsive design for the main layout of the website, ensuring proper functionality across mobile, tablet, and desktop devices.

## Changes
- Added media queries for mobile-first design:
   - Phone (<600px)
   - Tab-Port (<900px)
   - Tab-Land (<1200px)
   - Big-Desktop (1800px>)
- Updated the following components:
  - Header: Adjusted spacing, typography, and layout for smaller screens.
  - Footer: Fixed alignment issues and improved spacing for responsiveness.
  - Main content: Ensured text and images resize properly across breakpoints.
- Minor bug fixes for layout and positioning.

## Testing Instructions
1. Open the website on different devices or use developer tools to simulate:
   - Phone (<600px)
   - Tab-Port (<900px)
   - Tab-Land (<1200px)
   - Big-Desktop (1800px>)
2. Verify that the layout adapts correctly without breaking any elements.
3. Check buttons and links for proper alignment and functionality.

## Checklist
- [x] Code linted and formatted.
- [x] Changes tested across multiple devices and screen sizes.
- [x] No regressions or functionality breaks.
